narwhals/_dask/dataframe.py

```diff
@@ -96,17 +103,25 @@ def select(
         **named_exprs: IntoDaskExpr,
     ) -> Self:
         import dask.dataframe as dd  # ignore-banned-import
+        import dask_expr as de  # ignore-banned-import
+        import pandas as pd  # ignore-banned-import
 
         if exprs and all(isinstance(x, str) for x in exprs) and not named_exprs:
             # This is a simple slice => fastpath!
             return self._from_native_frame(self._native_frame.loc[:, exprs])
 
+        n_modifies_index = sum(
+            getattr(e, "_modifies_index", 0)
+            for e in list(exprs) + list(named_exprs.values())
+        )
+        if n_modifies_index > 1:
+            msg = "Found multiple expressions that modify the index"
+            raise ValueError(msg)
```

Comment:
TODO: Logic should be more complex:

- If total number of exprs is 1, then we can allow to modify the index
- If more than 1 exprs are present, then this should be banned as in `with_columns`